src/js/background.js

```diff
@@ -170,21 +170,44 @@ Badger.prototype = {
   updateTabList: function(){
     // Initialize the tabData/frames object if it is falsey
     this.tabData = this.tabData || {};
-    var self = this;
-    chrome.tabs.query({currentWindow: true, status: 'complete'}, function(tabs){
-      for(var i = 0; i < tabs.length; i++){
-        var tab = tabs[i];
-        self.tabData[tab.id] = {
-          frames: {
-            0: {
-              parent: -1,
-              url: tab.url
-            }
-          },
-          trackers: {}
-        };
+    let self = this;
+    chrome.tabs.query({}, tabs => {
+      tabs.forEach(tab => {
+        self.recordFrame(tab.id, 0, -1, tab.url);
+      });
+    });
+  },
+
+  /**
+   * Generate representation in internal data structure for frame
+   *
+   * @param tabId ID of the tab
+   * @param frameId ID of the frame
+   * @param parentFrameId ID of the parent frame
+   * @param frameUrl The url of the frame
+   */
+  recordFrame: function(tabId, frameId, parentFrameId, frameUrl) {
+    let self = this;
```

Comment:
Ah, never mind, we have a callback below, might need to bind context without the assignment. Let's just keep going with this. I'll remove these two comments now.